client.go

```diff
@@ -30,11 +30,18 @@ import (
 // GetEndpoints returns the available endpoint descriptions for the server.
 func GetEndpoints(ctx context.Context, endpoint string, opts ...Option) ([]*ua.EndpointDescription, error) {
 	opts = append(opts, AutoReconnect(false))
-	c := NewClient(endpoint, opts...)
+	c, err := NewClient(endpoint, opts...)
+	if err != nil {
+		return nil, err
+	}
 	if err := c.Dial(ctx); err != nil {
 		return nil, err
 	}
-	defer c.Close(ctx)
+	defer func(c *Client, ctx context.Context) {
```

Comment:
Let's not do this since there isn't anything we can do with that error anyway. If we really want to log it we should add something to the `Close` method.

Reply:
ok got a linter warning for that. Should I don't touch it at all or replace it with `_ = c.Close(ctx)`?

Reply:
![image](https://github.com/gopcua/opcua/assets/1769155/2d6f7eb3-9f8d-41fe-b47a-db14ce8ff37b)

Reply:
Please no. Fix your linter. This is a cargo cult being pushed on the Go community by people who want to change the Go code style through the back door. There is zero readability advantage and that's the main thing we are trying to achieve. It just adds unnecessary clutter.